base/src/com/google/idea/blaze/base/dependencies/ExternalFileProjectManagementHelper.java

```diff
@@ -167,42 +168,56 @@ public EditorNotificationPanel createNotificationPanelForLegacySync(VirtualFile
       return null;
     }
 
-    boolean addTargets = !alreadyBuilt && !AddSourceToProjectHelper.autoDeriveTargets(project);
-    ListenableFuture<List<TargetInfo>> targetsFuture =
-        addTargets
-            ? AddSourceToProjectHelper.getTargetsBuildingSource(context)
-            : Futures.immediateFuture(ImmutableList.of());
-    if (targetsFuture == null) {
-      return null;
-    }
-
-    EditorNotificationPanel panel =
-        createPanel(
-            vf,
-            p ->
-                p.createActionLabel(
-                    "Add file to project",
-                    () -> {
-                      AddSourceToProjectHelper.addSourceToProject(
-                          project, context.workspacePath, inProjectDirectories, targetsFuture);
-                      EditorNotifications.getInstance(project).updateNotifications(vf);
-                    }));
-    panel.setVisible(false); // starts off not visible until we get the query results
+    if(!AddSourceToProjectHelper.autoDeriveTargets(project)) {
+      boolean addTargets = !alreadyBuilt && !AddSourceToProjectHelper.autoDeriveTargets(project);
+      ListenableFuture<List<TargetInfo>> targetsFuture =
+              addTargets
+                      ? AddSourceToProjectHelper.getTargetsBuildingSource(context)
+                      : Futures.immediateFuture(ImmutableList.of());
+      if (targetsFuture == null) {
+        return null;
+      }
+      EditorNotificationPanel panel =
+              createPanel(
```

Comment:
It seems to me that some regions have formatting issues. I believe the indent should be 4 spaces instead of 8.